packages/amplify-e2e-core/src/utils/credentials-rotator.ts

```diff
@@ -0,0 +1,57 @@
+import { AssumeRoleCommand, STSClient } from '@aws-sdk/client-sts';
+import { fromContainerMetadata } from '@aws-sdk/credential-providers';
+import { generateRandomShortId, TEST_PROFILE_NAME } from './index';
+import * as ini from 'ini';
+import * as fs from 'fs-extra';
+import { pathManager } from '@aws-amplify/amplify-cli-core';
+
+const refreshCredentials = async (roleArn: string) => {
+  try {
+    const client = new STSClient({
+      // Use CodeBuild role to assume test account role. I.e. don't read credentials from process.env
+      credentials: fromContainerMetadata(),
+    });
+    const sessionName = `testSession${generateRandomShortId()}`;
+    const command = new AssumeRoleCommand({
+      RoleArn: roleArn,
+      RoleSessionName: sessionName,
+      DurationSeconds: 3600,
+    });
+    const response = await client.send(command);
+
+    const profileName = TEST_PROFILE_NAME;
+    const credentialsContents = ini.parse((await fs.readFile(pathManager.getAWSCredentialsFilePath())).toString());
+    credentialsContents[profileName] = credentialsContents[profileName] || {};
+    credentialsContents[profileName].aws_access_key_id = response.Credentials.AccessKeyId;
+    credentialsContents[profileName].aws_secret_access_key = response.Credentials.SecretAccessKey;
+    credentialsContents[profileName].aws_session_token = response.Credentials.SessionToken;
+    process.env.AWS_ACCESS_KEY_ID = response.Credentials.AccessKeyId;
+    process.env.AWS_SECRET_ACCESS_KEY = response.Credentials.SecretAccessKey;
+    process.env.AWS_SESSION_TOKEN = response.Credentials.SessionToken;
+    await fs.writeFile(pathManager.getAWSCredentialsFilePath(), ini.stringify(credentialsContents));
+    console.log('Test profile credentials refreshed');
+  } catch (e) {
+    console.error('Test profile credentials request failed');
+    console.error(e);
+  }
+};
+
+let isRotationScheduled = false;
+
+export const tryScheduleCredentialRefresh = () => {
+  if (!process.env.IS_AMPLIFY_CI || !process.env.TEST_ACCOUNT_ROLE || isRotationScheduled) {
```

Comment:
isn't this condition always false in CI?

Reply:
No. Logs here

```
Please migrate your code to use AWS SDK for JavaScript (v3).
--
103 | For more information, check the migration guide at https://a.co/7PzMCcy
104 | (Use `node --trace-warnings ...` to show where the warning was created)
105 | console.log
106 | Test profile credentials refresh was scheduled
107 |  
108 | at tryScheduleCredentialRefresh (../amplify-e2e-core/src/utils/credentials-rotator.ts:56:11)
109 |  
110 | console.log
111 | CLI_REGION set to: us-east-2. Overwriting AWS_REGION and AWS_DEFAULT_REGION
112 |  
113 | at src/setup-tests.ts:24:13
114 |  
115 | (node:531) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.
116 |  
117 | Please migrate your code to use AWS SDK for JavaScript (v3).
118 | For more information, check the migration guide at https://a.co/7PzMCcy
119 | (Use `node --trace-warnings ...` to show where the warning was created)
120 | console.log
121 | /tmp/amplify-e2e-tests/ad91ed979b5745deb278_859814
122 |  
123 | at createNewProjectDir (../amplify-e2e-core/src/index.ts:126:11)
124 |  
125 | console.log
126 | Test profile credentials refreshed
127 |  
128 | at refreshCredentials (../amplify-e2e-core/src/utils/credentials-rotator.ts:32:13)
129 |  
130 | console.log
131 | Test profile credentials refreshed
132 |  
133 | at refreshCredentials (../amplify-e2e-core/src/utils/credentials-rotator.ts:32:13)
134 |  
135 | console.log
136 | Test profile credentials refreshed
137 |  
138 | at refreshCredentials (../amplify-e2e-core/src/utils/credentials-rotator.ts:32:13)
139 |  
140 | console.log
141 | Skipping cloud deletion since we are in CI, and cleanup script will delete this stack in cleanup step.
142 |  
143 | at src/__tests__/transformer-migrations/searchable-migration.test.ts:43:15
144 |  
145 | PASS src/__tests__/transformer-migrations/searchable-migration.test.ts (3469.739 s)
146 | transformer model searchable migration test
147 | ✓ migration of searchable directive - search should return expected results (3442393 ms)
```

https://us-east-1.console.aws.amazon.com/codesuite/codebuild/671107461633/projects/AmplifyCLI-E2E-Testing/build/AmplifyCLI-E2E-Testing%3A554fff6b-76b0-4ad8-8d99-b65ad2f37934?region=us-east-1